internal/metric/prometheus.go

```diff
@@ -118,18 +119,16 @@ func checkReady(ctx context.Context, api promv1.API, t time.Time) (time.Time, er
 			}
 		}
 
-		if target.LastScrape.Before(t) {
+		// Ensure we have done an additional scrape since completion time
+		if target.LastScrape.Before(t.Add(scrapeInterval)) {
 			return t, &CaptureError{
 				Message:    "waiting for final scrape",
 				Address:    target.ScrapeURL,
 				RetryAfter: scrapeInterval,
 			}
 		}
 
-		// TODO This should use `target.LastScrapeDuration` once it is available
-		if ls := target.LastScrape.Add(scrapeTimeout); ls.After(lastScrape) {
-			lastScrape = ls
-		}
+		lastScrape = target.LastScrape
```

Comment:
Same as above, `target.LastScrape` is the start time of scrape, not the finish time. This needs to be the _end time_ of the scrape which we can only approximate with the scrape timeout.

Reply:
I was hoping we would wash over those details by including an additional scrape interval on top of completion time. As long as last scrape was after completion + scrape interval, we should be covered ( under the assumption scrape timeout must be less than scrape interval ).
So by the time we get to this point, lastScrape should be at least 1 interval past completion time.

Reply:
Yeah, just realized that was what was happening as soon as I stopped thinking about it.